### PR TITLE
Add sections to gateway helm chart for config

### PR DESCRIPTION
**Pull request by rani-sharim**
Added three sections to the helm chart:

To configure the defaults section:
defaultsConf:
  defaultSamlAccessId:
  defaultOidcAccessId:
  defaultEncryptionKey:
  defaultSecretLocation:

To configure the cache section:
cachingConf:
  enabled: <true/false>
  cacheTTL:
  proActiveCaching:
     enabled:  <true/false>
     minimumFetchingTime:
     cacheDumpInterval:

To configure the logand section:
logandConf:
   <logand file in string format>

### CODE REVIEW

File: charts/akeyless-api-gateway/values.yaml [hunk 1]
```diff
@@ -149,6 +149,33 @@ akeylessUserAuth:
...
+  # defaultSamlAccessId:
+  # defaultOidcAccessId:
+  # defaultEncryptionKey:
+  # defaultSecretLocation:
+
+
+# Specifies an initial configuration for the caching section
+cachingConf:
+  # cacheEnable: <true/false>
+  # cacheTTL:
```
Comment (renanaAkeyless) [inline charts/akeyless-api-gateway/values.yaml:164]: Add description to cacheTTL like: ttl in minutes
also proActiveCacheMinimumFetchingTime
and proActiveCacheDumpInterval

File: charts/akeyless-api-gateway/values.yaml [hunk 2]
```diff
@@ -149,6 +149,33 @@ akeylessUserAuth:
...
+  # proActiveCacheEnable:  <true/false>
+  # proActiveCacheMinimumFetchingTime:
+  # proActiveCacheDumpInterval:
+
+
+# Specifies an initial configuration for logand (log forwarding)
+# for example:
+# 
+# logandConf: |-
+#   target_syslog_host="127.0.0.1:514"
```
Comment (renanaAkeyless) [inline charts/akeyless-api-gateway/values.yaml:174]: enable="true"

Comment (renanaAkeyless) [reply charts/akeyless-api-gateway/values.yaml:174]: add link to doc:
 https://docs.akeyless.io/docs/ssh-log-forwarding#syslog-configuration

File: charts/akeyless-api-gateway/values.yaml (hunk 1 above)
Comment (rani-sharim) [reply charts/akeyless-api-gateway/values.yaml]: Done

File: charts/akeyless-api-gateway/values.yaml (hunk 2 above)
Comment (rani-sharim) [reply charts/akeyless-api-gateway/values.yaml:178]: Done

File: charts/akeyless-api-gateway/values.yaml [hunk 3]
```diff
@@ -149,6 +149,52 @@ akeylessUserAuth:
...
+  ## Default OIDC Access ID to be used for initial WebUI login
+  # defaultOidcAccessId:
+  ## This Default Encryption Key will be selected when creating the following items: Static Secrets, Dynamic Secret Producers and Secret Migration Configurations
+  # defaultEncryptionKey:
+  ## The location of the default path to save secrets
+  # defaultSecretLocation:
+
+
+# Specifies an initial configuration for the caching section
+cachingConf: {}
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/values.yaml:168]:
```suggestion
cachingConf: 
  enabled: false
```

File: charts/akeyless-api-gateway/values.yaml [hunk 4]
```diff
@@ -149,6 +149,52 @@ akeylessUserAuth:
...
+    
+  # cacheEnable: true
+  ## The amount of time during which a secret should be cached in minutes
+  # cacheTTL: 60
+
+  # proActiveCacheEnable: false
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/values.yaml:174]:
```suggestion
  # proActiveCaching:
     enabled: false 
```

File: charts/akeyless-api-gateway/values.yaml [hunk 5]
```diff
@@ -149,6 +149,52 @@ akeylessUserAuth:
...
+  
+  ## The minimum fetching interval in minutes (used to avoid fetching a secret too many times in a given timeframe)
+  # proActiveCacheMinimumFetchingTime: 5
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/values.yaml:177]: I would enter all of the proactive chaching values  indented under "proActiveCaching" like this:

proActiveCaching:
  minimumFetchingTime: 5
  dumpInterval: 1

File: charts/akeyless-api-gateway/templates/deployment.yaml [hunk 6]
```diff
@@ -124,6 +144,47 @@ spec:
...
+          {{- if .Values.defaultsConf.defaultEncryptionKey }}
+            - name: DEFAULT_ENCRYPTION_KEY
+              value: {{ .Values.defaultsConf.defaultEncryptionKey }}
+          {{- end }}
+           {{- if .Values.defaultsConf.defaultSecretLocation }}
+            - name: DEFAULT_SECRET_LOCATION
+              value: {{ .Values.defaultsConf.defaultSecretLocation }}
+          {{- end }}
+          ## caching section
+          {{- if .Values.cachingConf.cacheEnable }}
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/deployment.yaml:166]:
```suggestion
          {{- if .Values.cachingConf.enabled }}
```

File: charts/akeyless-api-gateway/templates/deployment.yaml [hunk 7]
```diff
@@ -124,6 +144,47 @@ spec:
...
+            - name: CACHE_ENABLE
+              value: "{{ .Values.cachingConf.cacheEnable }}"
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/deployment.yaml:168]:
```suggestion
              value: "{{ .Values.cachingConf.enabled }}"
```

File: charts/akeyless-api-gateway/templates/deployment.yaml [hunk 8]
```diff
@@ -124,6 +144,47 @@ spec:
...
+          {{- end }}
+          {{- if .Values.cachingConf.cacheTTL }}
+            - name: CACHE_TTL
+              value: "{{ .Values.cachingConf.cacheTTL }}"
+          {{- end }}
+          {{- if .Values.cachingConf.proActiveCacheEnable }}
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/deployment.yaml:174]:
```suggestion
          {{- if .Values.cachingConf.proActiveCaching.enabled }}
```

File: charts/akeyless-api-gateway/templates/deployment.yaml [hunk 9]
```diff
@@ -124,6 +144,47 @@ spec:
...
+            - name: PROACTIVE_CACHE_ENABLE
+              value: "{{ .Values.cachingConf.proActiveCacheEnable }}"
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/deployment.yaml:176]:
```suggestion
              value: "{{ .Values.cachingConf.proActiveCaching.enabled }}"
```

File: charts/akeyless-api-gateway/templates/deployment.yaml [hunk 10]
```diff
@@ -124,6 +144,47 @@ spec:
...
+          {{- end }}
+          {{- if .Values.cachingConf.proActiveCacheMinimumFetchingTime }}
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/deployment.yaml:178]:
```suggestion
          {{- if .Values.cachingConf.proActiveCaching.minimumFetchingTime }}
```

File: charts/akeyless-api-gateway/templates/deployment.yaml [hunk 11]
```diff
@@ -124,6 +144,47 @@ spec:
...
+            - name: PROACTIVE_CACHE_MINIMUM_FETCHING_TIME
+              value: "{{ .Values.cachingConf.proActiveCacheMinimumFetchingTime }}"
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/deployment.yaml:180]:
```suggestion
              value: "{{ .Values.cachingConf.proActiveCaching.minimumFetchingTime }}"
```

File: charts/akeyless-api-gateway/templates/deployment.yaml [hunk 12]
```diff
@@ -124,6 +144,47 @@ spec:
...
+          {{- end }}
+          {{- if .Values.cachingConf.proActiveCacheDumpInterval }}
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/deployment.yaml:182]:
```suggestion
          {{- if .Values.cachingConf.proActiveCaching.dumpInterval }}
```

File: charts/akeyless-api-gateway/templates/deployment.yaml [hunk 13]
```diff
@@ -124,6 +144,47 @@ spec:
...
+            - name: PROACTIVE_CACHE_DUMP_INTERVAL
+              value: "{{ .Values.cachingConf.proActiveCacheDumpInterval }}"
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/deployment.yaml:184]:
```suggestion
              value: "{{ .Values.cachingConf.proActiveCaching.dumpInterval }}"
```

File: charts/akeyless-api-gateway/values.yaml (hunk 3 above)
Comment (rani-sharim) [reply charts/akeyless-api-gateway/values.yaml]: done

File: charts/akeyless-api-gateway/values.yaml (hunk 4 above)
Comment (rani-sharim) [reply charts/akeyless-api-gateway/values.yaml]: done

File: charts/akeyless-api-gateway/values.yaml (hunk 5 above)
Comment (rani-sharim) [reply charts/akeyless-api-gateway/values.yaml]: done

File: charts/akeyless-api-gateway/templates/deployment.yaml (hunk 6 above)
Comment (rani-sharim) [reply charts/akeyless-api-gateway/templates/deployment.yaml]: done

File: charts/akeyless-api-gateway/templates/deployment.yaml (hunk 7 above)
Comment (rani-sharim) [reply charts/akeyless-api-gateway/templates/deployment.yaml]: done

File: charts/akeyless-api-gateway/templates/deployment.yaml (hunk 8 above)
Comment (rani-sharim) [reply charts/akeyless-api-gateway/templates/deployment.yaml]: done

File: charts/akeyless-api-gateway/templates/deployment.yaml (hunk 10 above)
Comment (rani-sharim) [reply charts/akeyless-api-gateway/templates/deployment.yaml]: done

File: charts/akeyless-api-gateway/templates/deployment.yaml (hunk 11 above)
Comment (rani-sharim) [reply charts/akeyless-api-gateway/templates/deployment.yaml]: done

File: charts/akeyless-api-gateway/templates/deployment.yaml (hunk 12 above)
Comment (rani-sharim) [reply charts/akeyless-api-gateway/templates/deployment.yaml]: done

File: charts/akeyless-api-gateway/templates/deployment.yaml (hunk 13 above)
Comment (rani-sharim) [reply charts/akeyless-api-gateway/templates/deployment.yaml]: done

**Review by niv-ez — APPROVED** [review]

**Pull request merged**